### PR TITLE
v9_changes_server_pod_failure

### PR DESCRIPTION
**Pull request by ApurwaMohite**
Signed-off-by: ApurwaMohite <55824145+ApurwaMohite@users.noreply.github.com>

# Problem Statement
- Problem statement

# Design
-  For Bug, Describe the fix here.
-  For Feature, Post the link for design

# Coding
   Checklist for Author
-  [ ] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
-  [ ] New/Affected tests are executed on Latest Build
-  [ ] Attach test execution logs
-  [ ] Collection tested and no collection error introduced (`pytest --local True --collect-only`)

# Review Checklist 
  Checklist for Author
- [ ] JIRA number/GitHub Issue added to PR
- [ ] PR is self reviewed
- [ ] Jira and state/status is updated and JIRA is updated with PR link
- [ ] Check if the description is clear and explained

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
-  [ ] If change in any common function, make sure to update all calls and execute all affected tests.

# Documentation
  Checklist for Author
-  [ ] Changes done to ReadMe / WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

File: tests/ha/single_pod_failure/test_server_pod_failure.py [hunk 1]
```diff
@@ -76,7 +76,6 @@ def setup_class(cls):
         cls.deploy_lc_obj = ProvDeployK8sCortxLib()
         cls.s3_clean = cls.test_prefix = cls.multipart_obj_path = None
         cls.s3acc_name = cls.s3acc_email = cls.bucket_name = cls.object_name = None
-        cls.restore_pod = cls.deployment_backup = cls.deployment_name = cls.restore_method = None
```
Comment (rahul-hatwar) [inline tests/ha/single_pod_failure/test_server_pod_failure.py:79]: line 67,68 not required

Comment (ApurwaMohite) [reply tests/ha/single_pod_failure/test_server_pod_failure.py:79]: not required

File: tests/ha/single_pod_failure/test_server_pod_failure.py [hunk 2]
```diff
@@ -174,15 +176,34 @@ def test_degraded_reads_safe_server_pod_shutdown(self):
         assert_utils.assert_true(resp[0], resp[1])
         LOGGER.info("Step 2: Performed READs and verified DI on the written data")
 
-        LOGGER.info("Step 3: Shutdown random server pod safely by making replicas=0 and verify "
-                    "cluster & remaining pods status")
+        num_replica = 0
+        LOGGER.info("Get pod to be deleted")
```
Comment (rahul-hatwar) [inline tests/ha/single_pod_failure/test_server_pod_failure.py:180]: Is is better to specify name of pod - data or server or data/server

Comment (ApurwaMohite) [reply tests/ha/single_pod_failure/test_server_pod_failure.py:180]: We are printing the pod name on line 186.

File: tests/ha/single_pod_failure/test_server_pod_failure.py [hunk 3]
```diff
@@ -286,23 +308,58 @@ def test_reads_during_server_pod_down(self):
...
+            self.set_name = resp[1][pod_name]['deployment_name']
+        elif set_type == const.REPLICASET:
+            self.deployment_name = resp[1][pod_name]['deployment_name']
         self.restore_pod = True
         self.restore_method = resp[1][pod_name]['method']
         assert_utils.assert_true(resp[0], "Cluster/Services status is not as expected")
         LOGGER.info("Step 3: Successfully shutdown server pod %s. Verified cluster and "
                     "services states are as expected & remaining pods status is online.", pod_name)
 
+        # TODO: Need confirmation from Rick on delete deployment method
```
[the remark this answers is not in the document]
Comment (ApurwaMohite) [reply tests/ha/single_pod_failure/test_server_pod_failure.py:345]: Sure, I will remove commented steps.

File: tests/ha/single_pod_failure/test_data_pod_failure.py [hunk 4]
```diff
@@ -2859,8 +2859,11 @@ def test_object_crud_pod_failure(self):
 
         num_replica = 0
         LOGGER.info("Get pod to be deleted")
-        pod_list = self.node_master_list[0].get_all_pods(pod_prefix=const.POD_NAME_PREFIX)
-        delete_pod = pod_list[-1]
+        sts_dict = self.node_master_list[0].get_sts_pods(pod_prefix=const.POD_NAME_PREFIX)
```
Comment (rahul-hatwar) [inline tests/ha/single_pod_failure/test_data_pod_failure.py:2862]: are we not adding this setup_method as we did in server failure file?

File: tests/ha/single_pod_failure/test_server_pod_failure.py [hunk 5]
```diff
@@ -201,6 +220,7 @@ def test_degraded_reads_safe_server_pod_shutdown(self):
     @pytest.mark.ha
     @pytest.mark.lc
     @pytest.mark.tags("TEST-39903")
+    @pytest.mark.skip("Skipped until CORTX-33759 is fixed")
```
Comment (rahul-hatwar) [inline tests/ha/single_pod_failure/test_server_pod_failure.py:223]: is this require?

File: tests/ha/single_pod_failure/test_server_pod_failure.py [hunk 6]
```diff
@@ -1783,6 +1849,7 @@ def test_mpu_safe_server_pod_down(self):
     @pytest.mark.ha
     @pytest.mark.lc
     @pytest.mark.tags("TEST-39915")
+    @pytest.mark.skip("Skipped until CORTX-33759 is fixed")
     @CTFailOn(error_handler)
```
Comment (rahul-hatwar) [inline tests/ha/single_pod_failure/test_server_pod_failure.py:1853]: please remove this, everywhere

File: tests/ha/single_pod_failure/test_server_pod_failure.py [hunk 7]
```diff
@@ -1783,6 +1849,7 @@ def test_mpu_safe_server_pod_down(self):
...
     def test_mpu_unsafe_server_pod_down(self):
```
Comment (rahul-hatwar) [inline tests/ha/single_pod_failure/test_server_pod_failure.py:1854]: shall we create new file for mpu, copy object and chunk test cases ? - test_server_pod_failure_api.py?

**Review by rahul-hatwar — APPROVED** [review]

File: tests/ha/single_pod_failure/test_server_pod_failure.py (hunk 7 above)
Comment (rahul-hatwar) [reply tests/ha/single_pod_failure/test_server_pod_failure.py:1854]: Please ignore this comments as we decided we are not doing these change in single failure file

**Review by sonalk0209 — APPROVED** [review]

File: tests/ha/single_pod_failure/test_server_pod_failure.py (hunk 5 above)
Comment (sonalk0209) [reply tests/ha/single_pod_failure/test_server_pod_failure.py:223]: yes this we would keep for now for CFT regression cycle but then once we implement some other method, we might either remove the test itself or modify this one with diff method

**Review by ankita-93 — APPROVED** [review]
Please take care of pylint warning and codacy issue. Rest LGTM
image

File: tests/ha/single_pod_failure/test_data_pod_failure.py (hunk 4 above)
Comment (ApurwaMohite) [reply tests/ha/single_pod_failure/test_data_pod_failure.py:2862]: We will add this to setup_method after doing changes in all the tests. For now I will keep this as is.

**Comment by ApurwaMohite** [comment]
[PR1778.log](https://github.com/Seagate/cortx-test/files/9216330/PR1778.log)
Collection log.

**Review by kachhwahadivya — APPROVED** [review]

**Pull request merged**